### PR TITLE
Unify error reporting and add error notification

### PR DESCRIPTION
**Pull request by Stypox**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Followup to #7142
- Created a `ErrorUtil` class that contains the three ways to report an error:
  - `openActivity`: just opens the error activity as always
  - `showSnackbar`: just opens the error snackbar as always
  - `createNotification`: creates a notification with a description containing the usual error message, and a title explaining that by tapping on the notification you can report the error. The notification disappears upon tapping, and is set to `IMPORTANCE_LOW` like other NewPipe notifications so that it does not make sounds.
- Correctly prevent exception from being serialized in `ErrorInfo`, eliminating the need for `EnsureExceptionSerializable` (see commit message for more info)
- Remove `PlayerErrorHandler` and instead always show error notification when an error occurs in the player. Also remove the related "report player error" setting.
- Add two new options in debug settings to show an error snackbar or create an error notification

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

To test I suggest using the "Crash the player" button from #7142 or triggering #7302 @litetex 

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/App.java [hunk 1]
```diff
@@ -246,9 +246,16 @@ private void initNotificationChannels() {
                 .setDescription(getString(R.string.hash_channel_description))
                 .build();
 
+        final NotificationChannelCompat errorReportChannel = new NotificationChannelCompat
+                .Builder(getString(R.string.error_report_channel_id),
+                NotificationManagerCompat.IMPORTANCE_LOW)
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/App.java:251]: Shouldn't there be an indentation

**Review by litetex — COMMENTED** [review]
Code LGTM only minor things.

Tested it a bit, seems to work as expected.

I'm glad that you managed to get rid of the ugly exception workaround 😄

Only one thing is a bit confusing:
When e.g. a player error happens, the player just stops and I don't notice the notification at the top (in particularly when the notification bar is already full due to my phones notch).
So I think showing a toast that says "Unexpected error occurred, tap notification for details" would be great. I would show the toast whenever a new error notification is shown.

File: app/src/main/java/org/schabi/newpipe/App.java (hunk 1 above)
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/App.java:251]: It also looks strange to me, but I just copied the formatting from the other notification channels above. Should I change them all?

**Comment by Stypox** [comment]
> When e.g. a player error happens, the player just stops and I don't notice the notification at the top (in particularly when the notification bar is already full due to my phones notch).
So I think showing a toast that says "Unexpected error occurred, tap notification for details" would be great. I would show the toast whenever a new error notification is shown.

I agree, I'll improve this

File: app/src/main/java/org/schabi/newpipe/App.java (hunk 1 above)
Comment (litetex) [reply app/src/main/java/org/schabi/newpipe/App.java:251]: I don't know.
Do it if you want :)

**Comment by Stypox** [comment]
You can also test the player error by trying to play in background this live: https://www.youtube.com/watch?v=Q3uwXbkheDo

**Comment by Stypox** [comment]
@litetex done :-)

File: app/src/main/java/org/schabi/newpipe/error/ErrorUtil.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import android.os.Build
+import android.view.View
+import android.widget.Toast
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import com.google.android.material.snackbar.Snackbar
+import org.schabi.newpipe.R
+
+class ErrorUtil {
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/error/ErrorUtil.kt:18]: I would appreciate some javadoc here, otherwise I think we are fine :)

Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/error/ErrorUtil.kt:29]: Yeah, I also wanted to add some javadoc but at the end I forgot. Done ;-)

Comment (litetex) [reply app/src/main/java/org/schabi/newpipe/error/ErrorUtil.kt:29]: Very nice. I love it 👍

**Review by litetex — APPROVED** [review]

**Comment by Mhowser** [comment]
I can't find the artifacts section you were describing earlier
Screenshot_20211229-164835

**Comment by litetex** [comment]
> The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

Click on CI and then scroll down ;)
Note: You have to be logged in otherwise GH won't show you the artifacts.

**Comment by Mhowser** [comment]
Ah, I was clicking the dropdown arrow, oops :P thanks for the help!

**Pull request merged**